python/tests/test_vast.py

```diff
@@ -20,59 +20,79 @@
 TEST_DB_DIR = "/tmp/test-vast-db"
 
 
-@pytest.fixture(autouse=True)
-async def vast_server():
+@pytest.fixture()
+async def endpoint():
     proc = await asyncio.create_subprocess_exec(
-        "vast", "-d", TEST_DB_DIR, "start", stderr=asyncio.subprocess.PIPE
+        "vast",
+        "-e",
+        ":0",
+        "-d",
+        TEST_DB_DIR,
+        "start",
+        "--print-endpoint",
+        stdout=asyncio.subprocess.PIPE,
+        stderr=asyncio.subprocess.PIPE,
     )
-    await asyncio.sleep(3)
-    yield
+    endpoint = await proc.stdout.readline()
+    endpoint = endpoint.decode("utf-8").strip()
+    logger.debug(f"{endpoint = }")
+    yield endpoint
     proc.terminate()
+    # TODO: kill if in case of timeout.
     await asyncio.wait_for(proc.wait(), 5)
     await asyncio.to_thread(shutil.rmtree, TEST_DB_DIR)
```

Comment:
if we run outside a container, should we also call this cleanup before starting the server in case the states gets dirty for some reason?

Reply:
Sounds reasonable.